app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -69,6 +68,8 @@ def index
         instance_ids = instances.pluck(:library_instance_id).reject(&:blank?)
         instance_ids += instances.where(:library_instance_id => nil)
         repositories = Repository.where(:id => instance_ids)
+      elsif params[:library] && params[:content_view_version_id] && params[:organization_id]
+        repositories = repositories.where(:environment_id => @organization.library.id, :content_view_version_id => params[:content_view_version_id])
```

Comment:
I'd think we'd be safer grabbing the organization based upon the content view version ID so that we are guaranteed to be using the same library instance.